### PR TITLE
Refactor splitComments

### PR DESCRIPTION
**Pull request by DzyubSpirit**
I think `splitComments` will benefit from refactoring. Mentioned in https://github.com/metarhia/metadoc/pull/9#discussion_r255421457

### CODE REVIEW

**Comment by SemenchenkoVitaliy** [comment]
Something like this will change behaviour
```js
// Function with no arguments
// Returns: <string>
const noArgumentsFunction = () => 'str';
```
I think it should be added to other tests as well.

File: lib/introspector.js [hunk 1]
```diff
@@ -45,42 +45,31 @@ const isNamedLine = line => NAMED_LINES.find(s => line.startsWith(s));
...
-      } else if (isNamedLine(lines[i])) {
-        paramStart = i;
-        paramEnd = i;
-        break;
-      }
-    } else if (paramsStarted && !lines[i].startsWith(' ')) {
-      paramEnd = i;
+  let i = 1;
+  let hasSignature = false;
+  while (i < lines.length && !lines[i].startsWith(' ')) {
```
Comment (SemenchenkoVitaliy) [inline lib/introspector.js]: If there are no parameters, but there are comments like `Returns`, `Throws` etc., they will go to the description. Something like this should help:
```suggestion
  while (i < lines.length && !lines[i].startsWith(' ') && !isNamedLine(lines[i])) {
```

File: lib/introspector.js [hunk 2]
```diff
@@ -45,42 +45,31 @@ const isNamedLine = line => NAMED_LINES.find(s => line.startsWith(s));
...
 
-  let signatureStart = paramStart;
-
-  for (let i = 1; i < paramStart; i++) {
-    if (lines[i].startsWith('Signature:')) {
-      signatureStart = i;
-      break;
-    }
+  let paramStart = i;
+  if (hasSignature) {
```
Comment (SemenchenkoVitaliy) [inline lib/introspector.js]: I think we can move `paramStart` after the loop and make it `const`
```js
  if (hasSignature) {
    i++;
    while (i < lines.length && lines[i].startsWith(' '.repeat(4))) i++;
  }
  const paramStart = i;
```

File: lib/introspector.js (hunk 1 above)
Comment (DzyubSpirit) [reply lib/introspector.js]: Thanks :) I'll add it to the test.

**Comment by SemenchenkoVitaliy** [comment]
ping @lundibundi @belochub @nechaido

File: lib/introspector.js [hunk 3]
```diff
@@ -45,42 +45,31 @@ const isNamedLine = line => NAMED_LINES.find(s => line.startsWith(s));
...
-  let signatureStart = paramStart;
-
-  for (let i = 1; i < paramStart; i++) {
-    if (lines[i].startsWith('Signature:')) {
-      signatureStart = i;
-      break;
-    }
+  if (hasSignature) {
+    i++;
+    while (i < lines.length && lines[i].startsWith(' '.repeat(4))) i++;
```
Comment (lundibundi) [inline lib/introspector.js]: Perhaps extract `' '.repeat(4)` to a variable (or constant)?

Comment (DzyubSpirit) [reply lib/introspector.js]: Done

**Comment by lundibundi** [comment]
Landed in https://github.com/metarhia/metadoc/commit/89f06113530c9ddc6f87441682c958266e1b89fd.

**Pull request closed without merging**